### DISCUSSION TITLE
Generate and HDR10+

### DISCUSSION THREAD

**Discussion by jerethi** [quietvoid/dovi_tool#61] [Q&A] [answered]
Am trying to understand proper use of the generate command. The readme seems to suggest that it can generate a binary RPU from an HDR10+ metadata JSON file. Does this mean, in effect, that HDR10+ metadata can be converted to DV?

**Comment by quietvoid** [comment, marked as answer]
Yes, partially.

**Comment by jerethi** [comment]
Interesting. I'd like to try this, but I'm unclear on the command line. Is the assets/generator_example.json command necessary?

**Reply by quietvoid** [reply]
The base of the metadata is using the generator configuration JSON.
See documentation here: https://github.com/quietvoid/dovi_tool/blob/main/generator.md

For HDR10+, that configuration is still necessary but does not have to be complete.
A decent default could be like this:

```json5
{
    "length": 1000,
    "target_nits": 1000,
    "level6": {
        "max_display_mastering_luminance": 1000,
        "min_display_mastering_luminance": 1,
        "max_content_light_level": 1000,
        "max_frame_average_light_level": 400
    }
}
```

The length will be replaced by what is in the HDR10+ metadata JSON.
You should adjust the level6 metadata accordingly.

**Comment by quietvoid** [comment]
There is also an example here: https://forum.makemkv.com/forum/viewtopic.php?p=113909#p113909

**Comment by jerethi** [comment]
Thank you! Incredibly helpful!
Just one last question - for the min_display_mastering_luminance variable, I have a value of 0.0050 cd/m2. How do I enter that since it's not an int?

**Reply by quietvoid** [reply]
It's multiplied by 10 000, so 50.

**Reply by jerethi** [reply]
Thanks - I _literally just read_ that on the makemkv forum. Sorry to bother, and thanks for your assistance.

**Comment by jerethi** [comment]
One last question. Can one use ffmpeg to pipe an HEVC file when injecting DV metadata? Or does it only work with a raw HEVC file?

**Reply by quietvoid** [reply]
Currently it doesn't work with piping.

**Reply by MarkPeloquin** [reply]
I for 1 would like to vote for piping to be made to work!

**Reply by ghost** [reply]
You can use mkvmerge for simply injecting the stream into mkv and you see the DV String correctly  with MediaInfo.

Simply Create a batch file in the same folder as mkvmerge.exe is and drop your video over this Batch or type inject.bat myvideo.mkv
The HEVC String with the DV Data included must be named "DV_Injected.hevc":

`"mkvmerge.exe" --output ^"%~dpn1_[DV INJECTED].mkv^" --no-video ^"^(^" ^"%~1^" ^"^)^" --language 0:und --compression 0:none ^"^(^" ^"%~dp1DV_Injected.hevc^" ^"^)^"`

Of course you can change names, this is only an example.

**Comment by jerethi** [comment]
What if there is no value for max_frame_average_light_level or max_content_light_level? Should those just be 0 then?

**Reply by quietvoid** [reply]
Yes
